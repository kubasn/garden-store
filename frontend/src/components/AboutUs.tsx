import React from "react";

const stores = [
  {
    name: "Gardena NY",
    location: "East Harlem, NY",
    tel: "412 543 654",
    email: "ny_store@gardena.com",
  },
  {
    name: "Gardena Chicago",
    location: "Maywood, Chicago",
    tel: "645 325 000",
    email: "chicago_store@gardena.com",
  },
  {
    name: "Gardena Nashville",
    location: "Midtown, Nashville",
    tel: "486 356 755",
    email: "nashville_store@gardena.com",
  },
];

const AboutUs = () => {
  return (
    <div className="h-full min-h-[calc(100vh-22rem)]">
      <div>
        <div className="flex flex-col  items-center capitalize">
          <h1 className="text-stone-700 text-2xl font-semibold">About us</h1>
          <div className="bg-stone-500 h-[2px] w-4/5 md:w-[200px] "></div>
        </div>

        <p className="w-4/5 md:w-1/2 text-center mx-auto mt-4">
          Welcome to our garden store! We are a family-owned business with a
          passion for all things green. Our team has a combined experience of
          over 20 years in the gardening industry, and we are dedicated to
          providing our customers with high-quality products and expert advice.
          We offer a wide selection of plants, seeds, tools, and accessories for
          all your gardening needs. Whether you're a seasoned pro or just
          starting out, we have something for everyone. In addition to our
          physical store, we also have a vibrant online community where garden
          enthusiasts can connect and share tips and ideas. We are constantly
          updating our inventory and adding new products, so make sure to check
          back often. Thank you for choosing our garden store.{" "}
          <span className="inline-block font-semibold">
            {" "}
            We hope to see you in-store or online soon!{" "}
          </span>
        </p>
      </div>
      <div className="mt-8">
        <div className="flex flex-col items-center text-stone-700">
          <h2 className="text-stone-700 text-xl font-semibold">Our stores</h2>
          <div className="h-[2px] w-4/5 md:w-[200px] bg-stone-500 "></div>
        </div>
        <div className="flex justify-around mt-6 ">
          {stores.map((store) => (
            <div>
              <p>{store.name}</p>
              <p>{store.location}</p>
              <p>{store.email}</p>
              <p>{store.tel}</p>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default AboutUs;
