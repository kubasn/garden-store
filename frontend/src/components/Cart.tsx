import React from "react";
import { MdKeyboardBackspace } from "react-icons/md";
import { motion } from "framer-motion";
import { BsFillTrashFill } from "react-icons/bs";
import { AiOutlineMinus, AiOutlinePlus } from "react-icons/ai";
import { useTypedSelector } from "../hooks/use-typed-selector";
import { useDispatch } from "react-redux";
import { actionType } from "../state/actionType";

const Cart = () => {
  const cart = useTypedSelector((state) => state.cart);
  const dispatch = useDispatch();

  const closeCart = () => {
    dispatch({
      type: actionType.SET_CART_SHOW,
      payload: { ...cart, show: !cart.show },
    });
  };

  return (
    <motion.div
      initial={{ opacity: 0, x: 200 }}
      animate={{ opacity: 1, x: 0 }}
      exit={{ opacity: 0, x: 200 }}
      className="fixed top-0 right-0 h-screen  w-full md:w-[23rem] bg-white drop-shadow-md flex flex-col z-[101] "
    >
      <div className="w-full flex items-center justify-between p-4 cursor-pointer">
        <motion.div whileTap={{ scale: 1.5 }}>
          <MdKeyboardBackspace
            className="text-gray-700 text-3xl"
            onClick={closeCart}
          />
        </motion.div>
        <p className="text-2xl font-semibold before:content-['Cart'] "> </p>
        <motion.p
          whileTap={{ scale: 1.2 }}
          className="flex  px-2 py-1 rounded-md text-rose-800 text-xl  "
        >
          <BsFillTrashFill className="relative right-1 top-[6px] text-lg" />{" "}
        </motion.p>
      </div>

      {/* botton */}
      <div className="w-full h-full rounded-t-[40px] border-y-[1px]  flex flex-col">
        <div className="w-full h-[340px] md:h-[42px] px-6 py-10 flex flex-col gap-3">
          {/* items */}
          {cart.items?.map((item) => {
            return (
              <div className="w-full p-1 px-2 border-y-2 border-gray-400 flex gap-3 items-center rounded-md ">
                <img
                  src="https://firebasestorage.googleapis.com/v0/b/gardenstore-7ffba.appspot.com/o/Images%2F1668589823421-tree1.png?alt=media&token=1febeb41-8619-4dd8-89d8-1cd1131b5d89"
                  className="w-20 h-20 max-w-[60px] rounded-full object-contain"
                />
                <div className=" flex flex-col">
                  <p className="text-lg text-gray-900 font-semibold">
                    Nice tree
                  </p>
                  <p className="text-lg text-gray-900 font-bold">85$</p>
                </div>

                <div className="group flex items-center gap-2 ml-auto cursor-pointer bg-gray-200 px-2 rounded-xl">
                  <motion.div whileTap={{ scale: 1.3 }} className="">
                    <AiOutlineMinus className=" " />
                  </motion.div>
                  <p className=" py-[1px] px-2 border-[1px] text-[0.8rem] ">
                    QTY: 1
                  </p>
                  <motion.div whileTap={{ scale: 1.3 }} className="">
                    <AiOutlinePlus className=" " />
                  </motion.div>
                </div>
              </div>
            );
          })}
        </div>
        {/* submit */}
      </div>
      <div className="bg-gray-200 text-gray-700 py-2 w-full text-center">
        ORDER SUMMARY
      </div>
      <div className="w-full flex-1 bg-gray-50  flex flex-col items-center justify-evenly px-8 pb-2">
        <div className="w-full flex items-center justify-between">
          <p className="text-gray-500 text-lg">Subtotal</p>
          <p className="text-gray-500 text-lg">8.5$</p>
        </div>
        <div className="w-full flex items-center justify-between">
          <p className="text-gray-500 text-lg">Delivery cost:</p>
          <p className="text-gray-500 text-lg">4.5$</p>
        </div>
        <div className="w-full border-b border-gray-600 my-2"></div>

        <div className="w-full flex items-center justify-between">
          <p className="text-gray-500 text-lg">Estimated total:</p>
          <p className="text-gray-500 text-lg">13$</p>
        </div>

        <motion.button
          whileTap={{ scale: 1.3 }}
          className="w-3/4 p-2 rounded-md bg-stone-700 hover:bg-stone-800 text-gray-50 text-lg my-2 hover:shadow-lg transition-all duration-150 ease-out"
        >
          Proceed to checkout
        </motion.button>
      </div>
    </motion.div>
  );
};

export default Cart;
