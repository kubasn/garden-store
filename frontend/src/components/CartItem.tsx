import React, { useState, useEffect } from "react";
import { motion } from "framer-motion";
import {
  AiOutlineHeart,
  AiOutlineMinus,
  AiOutlinePlus,
  AiTwotoneHeart,
} from "react-icons/ai";
import { useTypedSelector } from "../hooks/use-typed-selector";
import { actionType } from "../state/actionType";
import { useDispatch } from "react-redux";
import { MdOutlineCancel } from "react-icons/md";

interface CartItemProps {
  id: number;
  imageUrl: string;
  title: string;
  price: string;
  qty: number;
}
const CartItem: React.FC<CartItemProps> = ({
  id,
  imageUrl,
  title,
  price,
  qty,
}) => {
  const [itemQty, setItemQty] = useState<number>(qty);
  const [whishlist, setWhishlist] = useState<boolean>(false);
  let cart = useTypedSelector((state) => state.cart);
  const [items, setItems] = useState<any>(cart.items);
  const dispatch = useDispatch();
  let newItems = JSON.parse(JSON.stringify(cart.items));

  const addToWishlist = () => {
    setWhishlist(!whishlist);
  };

  const deleteItem = () => {};

  const updateCart = () => {
    console.log(items);
    localStorage.setItem("cart", JSON.stringify(items));
    dispatch({
      type: actionType.SET_CART_SHOW,
      payload: {
        ...cart,
        items: items,
      },
    });
  };

  useEffect(() => {
    updateCart();
  }, [items]);
  const changeQuantity = (sign: string, id: number) => {
    if (sign == "+") {
      newItems?.map((item: any, key: number) => {
        if (item.id == id) {
          item.qty += 1;
        }
      });
      setItemQty(itemQty + 1);
      setItems(newItems);
    }

    if (sign == "-" && itemQty > 0) {
      newItems?.map((item: any, key: number) => {
        if (item.id == id) {
          item.qty -= 1;
        }
      });
      setItemQty(itemQty - 1);
      setItems(newItems);
    }
  };

  return (
    <div
      id={String(id)}
      className=" w-4/5  md:w-full p-1 px-2 border-y-2 border-gray-400 flex gap-3 items-center rounded-md relative "
    >
      <img
        src={imageUrl}
        className="w-20 h-20 max-w-[60px] rounded-full object-contain"
      />
      <div className=" flex flex-col">
        <p className="text-lg text-gray-900 width-[2rem]">{title}</p>
        <p className="text-lg text-gray-900 font-bold">
          {parseFloat(price) * itemQty}$
        </p>
      </div>

      <div className="absolute right-1 bottom-2">
        <div className="group flex items-center gap-2 ml-auto cursor-pointer bg-gray-200 px-2 rounded-xl relative">
          {itemQty > 1 ? (
            <motion.div
              onClick={() => changeQuantity("-", id)}
              whileTap={{ scale: 1.3 }}
              className=""
            >
              <AiOutlineMinus className=" " />
            </motion.div>
          ) : (
            <AiOutlineMinus className="text-gray-400 " />
          )}
          <p className=" py-[1px] px-2 border-[1px] text-[0.8rem] ">
            QTY: {itemQty}
          </p>
          <motion.div
            onClick={() => changeQuantity("+", id)}
            whileTap={{ scale: 1.3 }}
            className=""
          >
            <AiOutlinePlus className=" " />
          </motion.div>
        </div>
      </div>
      <div className="text-xs flex gap-2 mt-1 absolute right-2 top-[0.2rem]">
        <AiOutlinePlus
          onClick={deleteItem}
          className="rotate-45 text-[17px] text-stone-800	hover:text-rose-700"
        />
        {whishlist ? (
          <AiTwotoneHeart
            onClick={addToWishlist}
            className=" text-[17px] 	text-rose-700 hover:text-rose-900"
          />
        ) : (
          <AiOutlineHeart
            onClick={addToWishlist}
            className=" text-[17px] text-stone-800	hover:text-rose-700"
          />
        )}
      </div>
    </div>
  );
};

export default CartItem;
