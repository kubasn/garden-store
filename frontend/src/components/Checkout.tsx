import React, { useEffect, useState } from "react";
import { useTypedSelector } from "../hooks/use-typed-selector";
import logo from "../images/logo.png";
import { motion } from "framer-motion";

interface cartType {
  category: string;
  description: string;
  id: number;
  imageUrl: string;
  price: string;
  qty: number;
  title: string;
}

const Checkout = () => {
  const cart = useTypedSelector((state) => state.cart);
  const user = useTypedSelector((state) => state.user);
  const [items, setItems] = useState<cartType[]>([]);
  let delivery = 10;
  const reduce = () => {
    let total = 0;
    if (cart.items != null)
      for (let val of cart.items) {
        total += parseFloat(val.price) * val.qty;
      }
    return total;
  };

  useEffect(() => {
    if (cart && cart.items !== null) setItems(cart.items);
  }, [cart]);

  return (
    <div className="h-full min-h-[calc(100vh-22rem)] flex flex-col gap-10">
      <div className="flex mx-auto">
        <p className=" text-stone-800 text-3xl uppercase">Checkout</p>
      </div>
      <div className="flex flex-col md:flex-row  justify-between ">
        {/* items */}
        <div className="md:w-1/2 w-[95%] mx-auto  ">
          {items.map((item) => (
            <div className="flex gap-2 border-b-[1px] last-of-type:border-none border-stone-400 pb-4 items-center relative ">
              <img className="w-20" src={item.imageUrl} />
              <div>
                <p className="">{item.title}</p>
                <p className="text-[12px] text-stone-400">Qty: {item.qty}</p>
              </div>
              <p className="absolute right-2 bottom-2 text-stone-600">
                {item.price} $
              </p>
            </div>
          ))}
        </div>
        {/* order summary */}
        <div className=" bg-white mx-auto w-4/5 md:w-1/3">
          <div className="bg-gray-200 py-2 pl-2 uppercase font-semibold">
            <p>Order summary</p>
          </div>

          <div className="pl-2">
            <p className="flex justify-between">
              <span>Subtotal</span>
              <span className="mr-4">{reduce()}</span>
            </p>
            <p className="flex justify-between">
              <span>Delivery</span>
              <span className="mr-4">{delivery} $</span>
            </p>
            <p className="flex justify-between mt-4 font-semibold ">
              <span>Estimated total</span>
              <span className="mr-4">{delivery + reduce()} $</span>
            </p>
          </div>
          <div className="w-4/5 h-[1px] bg-gray-200 my-4 m-auto"></div>
          <motion.button
            whileTap={{ scale: 1.1 }}
            className="w-4/5 m-auto flex py-1 justify-center border-[1px] text-white text-sm bg-sky-900"
          >
            TRADITIONAL PAYMENT
          </motion.button>
          <div className="flex items-center gap-2 w-[98%] m-auto ">
            <div className="w-4/5 h-[1px] bg-gray-200 my-4 m-auto"></div>
            <p className="text-gray-500">Or</p>
            <div className="w-4/5 h-[1px] bg-gray-200 my-4 m-auto"></div>
          </div>
          <motion.button
            whileTap={{ scale: 1.1 }}
            className=" flex border-[1px] border-black w-4/5 m-auto mb-4 rounded-sm"
          >
            <img src={logo} className="w-1/2 m-auto py-1  " alt="" />
          </motion.button>
        </div>
      </div>
    </div>
  );
};

export default Checkout;
