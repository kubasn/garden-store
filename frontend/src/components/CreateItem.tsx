import React, { ChangeEvent, useState } from "react";
import { RiPlantFill } from "react-icons/ri";
import { GiPriceTag } from "react-icons/gi";
import { AiOutlineCloudDownload, AiFillDelete } from "react-icons/ai";
import { MdDescription } from "react-icons/md";
import { motion } from "framer-motion";
import { categories } from "../backend";
import Loader from "./UI/Loader";
import {
  deleteObject,
  getDownloadURL,
  ref,
  uploadBytesResumable,
} from "firebase/storage";
import { storage } from "../firebase.config";
import { getItems, saveItem } from "./utils/functionsFirebase";
import { useDispatch } from "react-redux";
import { actionType } from "../state/actionType";

interface isItem {
  title: string;
  price: string;
  description: string;
  category: string;
  image: string;
}

const CreateItem = () => {
  const [loading, setLoading] = useState(false);
  const [fields, setFields] = useState(false);
  const [alert, setAlert] = useState("");
  const [msg, setMsg] = useState("");
  const dispatch = useDispatch();

  const [item, setItem] = useState<isItem>({
    title: "",
    price: "",
    description: "",
    category: "",
    image: "",
  });
  console.log(item);

  const fetchData = async () => {
    await getItems().then((data) => {
      console.log(data);
      dispatch({
        type: actionType.SET_ITEMS,
        payload: data,
      });
    });
  };

  const changeFunction = (
    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>,
    type: string
  ) => {
    setItem({ ...item, [type]: e.target.value });
    console.log(item);
  };

  const uploadImage = (e: ChangeEvent<HTMLInputElement>) => {
    setLoading(true);
    const imageFile = e.target.files![0];
    const storageRef = ref(storage, `Images/${Date.now()}-${imageFile.name}`);
    const uploadTask = uploadBytesResumable(storageRef, imageFile);
    uploadTask.on(
      "state_changed",
      (snapshot) => {
        const progress =
          (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
      },
      (error) => {
        console.log(error);
        setMsg("Error while uploading, try again!");
        setFields(true);
        setAlert("danger");

        setTimeout(() => {
          setLoading(false);

          setFields(false);
        }, 5000);
      },
      () => {
        getDownloadURL(uploadTask.snapshot.ref).then((downloadURL) => {
          setItem({ ...item, image: downloadURL });
          console.log("ello");
          setLoading(false);
          setFields(true);
          setMsg("Image uploaded");
          setAlert("success");

          setTimeout(() => {
            setLoading(false);

            setFields(false);
          }, 5000);
        });
      }
    );
  };
  const deleteImage = () => {
    setLoading(true);
    const deleteRef = ref(storage, item.image);
    deleteObject(deleteRef).then(() => {
      item.image = "";
      setLoading(false);
      setFields(true);
      setMsg("Image deleted successfully");
      setAlert("success");
      setTimeout(() => {
        setFields(false);
      }, 400);
    });
  };
  const saveDetails = () => {
    setLoading(true);
    try {
      if (
        !item.title ||
        !item.category ||
        !item.description ||
        !item.image ||
        !item.price
      ) {
        console.log(item);
        setFields(true);
        setMsg("Fields cannot be empty");
        setAlert("danger");

        setLoading(false);
        setTimeout(() => {
          setFields(false);
          setLoading(false);
        }, 4000);
      } else {
        const data = {
          id: Math.floor(Math.random() * 100000),
          title: item.title,
          price: item.price,
          description: item.description,
          category: item.category,
          imageUrl: item.image,
        };
        saveItem(data);

        setLoading(false);
        clearData();
        setFields(true);
        setMsg("Fields uploaded successfully");
        setAlert("success");

        setTimeout(() => {
          setFields(false);
        }, 400);
      }
    } catch (err) {
      setFields(true);
      setMsg("Error while uploading: Try again");
      setAlert("danger");

      setTimeout(() => {
        setFields(false);
        setLoading(false);
      }, 4000);
    }

    fetchData();
  };

  const clearData = () => {
    setItem({ title: "", price: "", description: "", category: "", image: "" });
  };

  return (
    <div className="h-auto min-h-screen w-full p-4 flex items-center justify-center">
      <div className="w-[90%] md:w-[75%] border border-gray-200 rounded-lg p-4">
        {fields && (
          <motion.p
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className={`w-full p-2 rounded-lg text-xl font-semibold text-center ${
              alert == "danger"
                ? "bg-red-300 text-red-700"
                : "bg-green-300 text-green-700"
            }`}
          >
            {msg}
          </motion.p>
        )}
        <div className="w-full py-2 border-b border-gray-300  flex flex-col  gap-2 ">
          {/* icon */}
          <div className="relative w-full mx-auto flex   justify-center gap-2">
            <div className="relative w-1/2">
              <RiPlantFill className="absolute left-[-25px] top-1 text-xl text-gray-700" />
              <input
                type="text"
                required
                value={item.title}
                onChange={(e) => changeFunction(e, "title")}
                placeholder="Insert title"
                className=" w-full h-full text-lg p-1 px-2 outline-none  text-gray-700 "
              />
            </div>
          </div>

          <div className="w-full flex justify-center">
            <select
              onChange={(e) => changeFunction(e, "category")}
              className="w-1/2  outline-none p-2 text-gray-500"
            >
              <option value="other" className="bg-white">
                Select category
              </option>
              {categories &&
                categories.map((item) => {
                  return (
                    <option
                      className="text-base border-0 outline-none text-gray-600"
                      value={item.urlParamName}
                    >
                      {item.name}
                    </option>
                  );
                })}
            </select>
          </div>
        </div>
        <div className="mt-2 group flex justify-center items-center flex-col border-2 border-dotted border-gray-300 w-full h-[240px] md:h-[480px] cursor-pointer rounded-md">
          {loading ? (
            <Loader />
          ) : (
            <>
              {!item.image ? (
                <label className="w-full h-full flex flex-col items-center justify-center cursor-pointer">
                  <div className="w-full h-full flex flex-col items-center justify-center gap-1">
                    <AiOutlineCloudDownload className="text-[50px] text-gray-500 hover:text-gray-700" />
                    <p className="text-gray-500">Click here to upload image</p>
                  </div>
                  <input
                    type="file"
                    name="uploadImg"
                    accept="image/*"
                    className="w-0 h-0"
                    onChange={uploadImage}
                  />
                </label>
              ) : (
                <div className="relative h-full">
                  <img
                    className="w-full h-full object-cover"
                    src={item.image}
                    alt="uploaded image"
                  />
                  <button
                    onClick={deleteImage}
                    className="absolute bottom-3 right-3 p-3 rounded-full bg-red-500 text-xl cursor-pointer outline-none hover:shadow-md transition-all ease-in-out duration-500 "
                  >
                    <AiFillDelete className="text-white" />
                  </button>
                </div>
              )}
            </>
          )}
        </div>
        <div className="mt-4">
          <div className="relative w-full mx-auto flex   justify-center gap-2">
            <div className="relative w-1/2">
              <GiPriceTag className="absolute left-[-25px] top-1 text-xl text-gray-700" />
              <input
                type="text"
                required
                value={item.price}
                onChange={(e) => changeFunction(e, "price")}
                placeholder="Insert price"
                className=" w-full h-full text-lg p-1 px-2  outline-none  text-gray-700 "
              />
            </div>
          </div>
          <div className="relative w-full mx-auto flex mt-2    justify-center gap-2">
            <div className="relative w-1/2">
              <MdDescription className="absolute left-[-25px] top-1 text-xl text-gray-700" />
              <textarea
                required
                value={item.description}
                onChange={(e) => changeFunction(e, "description")}
                placeholder="Description of item..."
                className=" w-full h-full text-lg p-1 px-2  outline-none  text-gray-700  "
              />
            </div>
          </div>
        </div>
        <button
          className=" w-1/2 md:w-1/5 flex mx-auto justify-center  py-1 rounded-sm hover:rounded-md   mt-4 text-white bg-emerald-600 hover:bg-emerald-700 transition-all ease-in-out duration-500"
          onClick={saveDetails}
        >
          SAVE
        </button>
      </div>
    </div>
  );
};

export default CreateItem;
