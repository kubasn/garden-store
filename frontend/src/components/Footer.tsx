import React from "react";
import { BsFacebook, BsInstagram, BsSnapchat, BsYoutube } from "react-icons/bs";

const Footer = () => {
  return (
    <div className="w-full h-[11rem] bg-stone-800 flex flex-col">
      <div className="flex gap-6 m-auto text-4xl text-white">
        <BsFacebook className="hover:text-blue-800 cursor-pointer" />
        <BsInstagram className="hover:text-rose-300 cursor-pointer" />
        <BsSnapchat className="hover:text-yellow-300 cursor-pointer" />
        <BsYoutube className="hover:text-rose-700 cursor-pointer" />
      </div>
      <span className="h-[1px] w-full bg-stone-100"></span>
      <div className="m-auto text-white">
        <p>Â© COPYRIGHT 2022 GARDENA STORE</p>
      </div>
    </div>
  );
};

export default Footer;
