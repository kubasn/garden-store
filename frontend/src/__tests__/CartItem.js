import { Provider } from "react-redux";
import { render, cleanup, screen } from "@testing-library/react";
import { createMockStore } from "redux-test-utils";
import { CartItem } from "../components";

afterEach(cleanup);

let item = {
  id: 1,
  imageUrl: "image.png",
  title: "Product Title",
  price: "$10",
  qty: 2,
};

describe("myComponent", () => {
  let store;
  beforeEach(() => {
    store = createMockStore({
      cart: {
        items: {
          id: 1,
          imageUrl: "image.png",
          title: "Product Title",
          price: "10",
          qty: 2,
        },
      },
      user: {
        uid: 123,
        name: "Mark",
      },
    });
  });

  it("renders correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CartItem />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("dispatches an action on button click", () => {
    const { getByText, getByRole, container } = render(
      <Provider store={store}>
        <CartItem {...item} />
      </Provider>
    );
    setTimeout(() => {}, 1000);
    screen.debug();

    const el = getByText(item.title);
    const title = getByText(new RegExp(item.title));
    // const price = getByText(new RegExp(item.price));
    const qty = getByText(new RegExp(item.qty));
  });
});
