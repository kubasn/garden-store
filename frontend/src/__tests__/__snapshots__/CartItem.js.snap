// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component should render with given state from Redux store 1`] = `
<div
  className=" w-4/5  md:w-full p-1 px-2 border-y-2 border-gray-400 flex gap-3 items-center rounded-md relative "
  id="undefined"
>
  <img
    className="w-20 h-20 max-w-[60px] rounded-full object-contain"
  />
  <div
    className=" flex flex-col"
  >
    <p
      className="text-lg text-gray-900 width-[2rem]"
    />
    <p
      className="text-lg text-gray-900 font-bold"
    >
      NaN
      $
    </p>
  </div>
  <div
    className="absolute right-1 bottom-2"
  >
    <div
      className="group flex items-center gap-2 ml-auto cursor-pointer bg-gray-200 px-2 rounded-xl relative"
    >
      <svg
        className="text-gray-400 "
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
        />
      </svg>
      <p
        className=" py-[1px] px-2 border-[1px] text-[0.8rem] "
      >
        QTY: 
      </p>
      <div
        className=""
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className=" "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          t="1551322312294"
          version="1.1"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
          />
          <path
            d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="text-xs flex gap-2 mt-1 absolute right-2 top-[0.2rem]"
  >
    <svg
      className="rotate-45 text-[17px] text-stone-800	hover:text-rose-700"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      t="1551322312294"
      version="1.1"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
      />
      <path
        d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
      />
    </svg>
    <svg
      className=" text-[17px] text-stone-800	hover:text-rose-700"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
      />
    </svg>
  </div>
</div>
`;

exports[`myComponent renders correctly 1`] = `
<DocumentFragment>
  <div
    class=" w-4/5  md:w-full p-1 px-2 border-y-2 border-gray-400 flex gap-3 items-center rounded-md relative "
    id="undefined"
  >
    <img
      class="w-20 h-20 max-w-[60px] rounded-full object-contain"
    />
    <div
      class=" flex flex-col"
    >
      <p
        class="text-lg text-gray-900 width-[2rem]"
      />
      <p
        class="text-lg text-gray-900 font-bold"
      >
        NaN$
      </p>
    </div>
    <div
      class="absolute right-1 bottom-2"
    >
      <div
        class="group flex items-center gap-2 ml-auto cursor-pointer bg-gray-200 px-2 rounded-xl relative"
      >
        <svg
          class="text-gray-400 "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
        <p
          class=" py-[1px] px-2 border-[1px] text-[0.8rem] "
        >
          QTY: 
        </p>
        <div
          class=""
        >
          <svg
            class=" "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            t="1551322312294"
            version="1.1"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
            />
            <path
              d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="text-xs flex gap-2 mt-1 absolute right-2 top-[0.2rem]"
    >
      <svg
        class="rotate-45 text-[17px] text-stone-800	hover:text-rose-700"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        t="1551322312294"
        version="1.1"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
        />
        <path
          d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
        />
      </svg>
      <svg
        class=" text-[17px] text-stone-800	hover:text-rose-700"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
